packages/core-data/src/types/menu-location.ts

```diff
@@ -0,0 +1,19 @@
+/**
+ * Internal dependencies
+ */
+import { Context } from './common';
+
+export interface MenuLocation< C extends Context > {
```

Comment:
The generic is unused here, is it necessary? If we're setting ourselves up for future additions to MenuLocation that might depend on the context, it'd be worth adding a comment to that effect on the interface.

Though I question whether that _should_ be done. I suppose it prevents future breaking changes for plugins that extend these types via interface merging, if this type ever did need the context in the future it'd require it. Additionally, the context could be used by a plugin to extend this type.

FWIW, I think we should keep the context for that final reason, to allow for using interface merging to add properties that do depend on the context; but let's add a comment to that effect to explain why there's an unused generic parameter.

Reply:
I thought we'd need the type parameter in order to use `Updatable` but apparently it's fine without it.

[proof is in the playground pudding](https://www.typescriptlang.org/play?#code/C4TwDgpgBAwsAeUC8UDkEAmBLYqoB80A3LCAd1QG4AoarAO2AgCcAzAQwGNoAFAewDOwADwwoEeE3oYBsBAD4oAb2pQ1UYDgA2EAFxQhzBgHMaAX1oMmbLtACyEegFdlq9ZuA79hk+dqhIKABRRhwQUXFJRxk5eEUUfiFRRUIHZxpqAOgAVTAMdmthABVIqRiQj3D0bFx5eKgSgDJlKCIWASw+en0ARigzDM4uoSh2ZGUzUdlc-MK0p3kadmoAehWoAD0Afloh+hGAI3GlSfZpvIKWYUSRapxUOpoD1fXt6iA)